drivers: i2s: i2s_mcux_sai: control MCLK direction with DT property

DerekSnell opened the pull request:
For the SAI peripheral, the MCLK signal input/output direction is independent from the TX or RX bit clocks directions (TCR2[BCD] and RCR2[BCD]). Introduces mclk-output property.

i2s_speed test passes on these platforms:
- frdm_mcxn947//cpu0
- mcx_n9xx_evk//cpu0 (using https://github.com/zephyrproject-rtos/zephyr/pull/87639)
- mimxrt1060_evk@B
- mimxrt1170_evk@A//cm7
- mimxrt1170_evk@B//cm7

Commit:
doc: releases: migration-guide-4.2: add mclk-output for nxp,mcux-i2s

Document new mclk-output property.

Signed-off-by: Derek Snell <derek.snell@nxp.com>

decsny commented on drivers/i2s/i2s_mcux_sai.c:
update copyright year

DerekSnell replied:
thanks @decsny , updated

Commit:
drivers: i2s_mcux_sai: control MCLK direction with DT property

For the SAI peripheral, the MCLK signal input/output direction is
independent from the TX or RX bit clocks directions (TCR2[BCD] and
RCR2[BCD]). Introduces mclk-output property.

Signed-off-by: Derek Snell <derek.snell@nxp.com>

DerekSnell commented:
rebased to enable merge
